CHANGELOG.rst

```diff
@@ -5,31 +5,25 @@
 
 Renku ``0.21.0`` brings the ability to handle expiring Gitlab tokens in user sessions and various bug fixes. 
 
-User-Facing Changes
-~~~~~~~~~~~~~~~~~~~
-
-**✨ Improvements**
-
-
-**🐞Bug Fixes**
-
-
 Internal Changes
 ~~~~~~~~~~~~~~~~
 
 **Improvements**
 
 * **Gateway**: Add endpoint for refreshing expiring Gitlab tokens
+* **Sessions**: Check LFS size and available disk space before cloning
```

Comment:
Depends what you mean by "before launching the session"? This does not happen before the jupyter server is created by a user. But it does happen before the jupyter server container in the pod starts. It happens before the init container that clones the project starts downloading data. So if the project is large and a lot of LFS data is being downloaded then the error message will appear very quickly rather than starting the download and erroring out only when the disk is full.

The error appears in the ui the same way any message for a failed session launch appears. The status of the session changes to failed and there is a little red exclamation mark that shows the message when you click it. Essentially the contents of the message and where the message is the same as when the sessions runs out of disk space now - this just makes it appear way faster. So now if you start a session that has data in LFS but you do not have enough disk space you will very likely see the error message in less than a minute, probably is 10-20 seconds.

Reply:
ok got it, but it doesn't prevent the launch. Does it keep the failing pod around?

Reply:
@rokroskar yes it does nothing to clean up the failing pod.

We can enable the automated failed pod culling and let that take care of it and others.

Reply:
I think that would be nice, otoh we usually learn something from being able to inspect those pods so 🤷